### PR TITLE
add fidoiot documentation

### PR DESCRIPTION
**Pull request by TomBarnes**
Signed-off-by: Tom Barnes <thomas.j.barnes@intel.com>

### CODE REVIEW

**Review by trbehera — DISMISSED** [review]
Look good.

File: docs/fidoiot/fidoiot-client-sdk-design.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+There are a few blocks and implementation to be either fully rewritten or modified to make the Client SDK compatible with the FIDO IOT draft  specification. Those changes are described in the following sections:
+
+### CBOR – COSE HAL Library (A1)
+
+This block implements the following functionality:
+
+1. CBOR Encode API – Uses the TinyCBOR library calls directly to encode data provided from upper layer to create CBOR data stream.  Data is provided back to SDO library to be sent using network layer.
+2. CBOR Decode API – Uses TinyCBOR library calls directly to decode received CBOR encoded data from servers.  Library calls may be used recursively.
+3. CBOR COSE APIs – Implements the COSE API requirement using the Crypto HAL API calls as well as using TinyCBOR APIs calls. Refer the the FIDO IOT draft specification for further details on API requirements.
```
Comment (DukeDavis12) [inline docs/fidoiot/fidoiot-client-sdk-design.md]: Please remove the redundant 'the'.

File: docs/fidoiot/fidoiot-client-sdk-design.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+### ServiceInfo Message Handling (A3)
+
+The FIDO IOT draft specification requires changes to the handling of serviceinfo messages. Handling of service info is done almost similar to previous specification; however, the repeat transmit, acknowledgement and DSI messages require full implementation in the SDO library. Unlike the current implementation,  service message must be part of the SDO library rather than a separate module.
+
+### Storage Implementation - CBOR Format (A4)
+
+Currently all manufacturer information and device secrets are stored on in persistent storage in JSON format. As new FIDO IOT draft specification uses CBOR as data format all data stored on storage need to be done using CBOR encoding. This will reduce confusion and restrict the user to use multiple data storage format.
+
+All storage file save and load APIs must be be rewritten using CBOR-COSE APIs to replace JSON encoding and decoding.
```
Comment (DukeDavis12) [inline docs/fidoiot/fidoiot-client-sdk-design.md]: Please remove the redundant 'be'.

File: docs/fidoiot/fidoiot-client-sdk-design.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+This layer must be adapted to the new FIDO IOT implementation. Each message API must be revisited and re-implemented as required to comply with the FIDO IOT message structure. This layer will extensively use CBOR-COSE HAL layer.
+
+### ServiceInfo Message Handling (A3)
+
+The FIDO IOT draft specification requires changes to the handling of serviceinfo messages. Handling of service info is done almost similar to previous specification; however, the repeat transmit, acknowledgement and DSI messages require full implementation in the SDO library. Unlike the current implementation,  service message must be part of the SDO library rather than a separate module.
+
+### Storage Implementation - CBOR Format (A4)
+
+Currently all manufacturer information and device secrets are stored on in persistent storage in JSON format. As new FIDO IOT draft specification uses CBOR as data format all data stored on storage need to be done using CBOR encoding. This will reduce confusion and restrict the user to use multiple data storage format.
```
Comment (DukeDavis12) [inline docs/fidoiot/fidoiot-client-sdk-design.md]: Consider adding a comma after Currently,

File: docs/fidoiot/fidoiot-client-sdk-design.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+All storage file save and load APIs must be be rewritten using CBOR-COSE APIs to replace JSON encoding and decoding.
+
+### Fido Message Handling 
+
+Following design is detail design of handling FIDO CBOR messages in Client SDK implementation.
```
Comment (DukeDavis12) [inline docs/fidoiot/fidoiot-client-sdk-design.md]: ..

File: docs/fidoiot/fidoiot-introduction.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+The [PRI-FIDOIOT](https://github.com/secure-device-onboard/pri-fidoiot) component has been added to the LF Edge Secure Device Onboard project.  It is a reference implementation of the specification working draft, and also implements the service modules required to exercise the complete system: Device, Manufacturer, Rendezvous Server, and Owner.  See the [readme](https://github.com/secure-device-onboard/pri-fidoiot/blob/master/README.md) in the PRI-FIDOIOT project repository for details on the implementation and how to get it set up and running.  See the [release notes](https://github.com/secure-device-onboard/release/releases) for PRI-FIDOIOT 0.3 for a summary of features and capabilities implemented (or not) in the 0.3 release.
+
+The Client SDK - FIDOIOT component will be added to the SDO project in the near future.  It will be based on the current SDO Client SDK.  A [design document](fidoiot-client-sdk-design.md) outlining the changes required to support the new FIDO protocols for devices as described in the current draft has been added to the SDO documentation.  Comments are welcome and can be submitted via the SDO documentation github repository.
```
Comment (DukeDavis12) [inline docs/fidoiot/fidoiot-introduction.md]: 1. in the near future --> soon.
2. github  --> GitHub.

File: docs/fidoiot/fidoiot-client-sdk-design.md (hunk 1 above)
Comment (DukeDavis12) [reply docs/fidoiot/fidoiot-client-sdk-design.md]: line 26

File: docs/fidoiot/fidoiot-client-sdk-design.md (hunk 2 above)
Comment (DukeDavis12) [reply docs/fidoiot/fidoiot-client-sdk-design.md]: line 44

File: docs/fidoiot/fidoiot-introduction.md (hunk 5 above)
Comment (TomBarnes) [reply docs/fidoiot/fidoiot-introduction.md]: @DukeDavis12 Thanks for your comments.  I will be pushing an update shortly.

**Review by DukeDavis12 — APPROVED** [review]
Looks Great!

**Comment by bprashan** [comment]
GitHub page layout looks good with below versions
mkdocs-1.1.2
mkdocs-material-5.5.12
mike-v0.5.5

File: docs/fidoiot/fidoiot-client-sdk-design.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+![FIDO IOT Message Handling](img/Slide4.PNG)
+
+1. CBOR–COSE HAL API Library will be used to decode and encode messages.
+2. Crypto HAL calls will be reutilized. If additional Crypto HAL APIs are required by CDDL they can be added as part of Crypto HAL layers.
+
+Figure 3. FIDO IOT Transmit Message Flowchart
+![FIDO IOT Transmit Message Flowchart](img/Slide5.PNG)
+
+Figure 4. FIDO IOT Receive Message Flowchart
+![FIDO IOT Receive Message Flowchart](img/Slide6.PNG)
```
Comment (prateekcr) [inline docs/fidoiot/fidoiot-client-sdk-design.md:60]: The images in both Figure 3 and 4 have the 'Yes'/'No' missing in the flow-diagram at a couple of places.

Comment (prateekcr) [reply docs/fidoiot/fidoiot-client-sdk-design.md:60]: Looks good otherwise.

**Review by trbehera — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Review by v-vasavi — APPROVED** [review]

File: docs/fidoiot/fidoiot-client-sdk-design.md (hunk 6 above)
Comment (TomBarnes) [reply docs/fidoiot/fidoiot-client-sdk-design.md:60]: Good catch - I will fix that in a future PR

**Pull request merged**